examples/hpdemo.cpp

```diff
@@ -0,0 +1,318 @@
+//                     hp refinement demo (NOT TO BE MERGED)
```

Comment:
Why not merge `hpdemo` into master? Also the `hpfem` miniapp from branch `p-refinement` works (in my branch `hpfem-var-order-space` from this branch). Why not include it? The unit test for variable order is good, but users might find an example or miniapp more illustrative.

Reply:
The demo is too simple, not really useful and has to use `ProlongToMaxOrder` at the moment, which would probably confuse users. There will be at least two more PRs, eventually we will probably have a proper example too. The hp miniapp is IMO not ready either, we should first get rid of the oscillations due to the relaxed mode (if you are using the miniapp and wouldn't mind helping experimenting with DG integrators tu suppress the oscillations, that would be a great help).